enhancements/network/bgp-ovn-kubernetes.md

```diff
@@ -0,0 +1,366 @@
+
+However, there is an alternative need to advertise pod IPs on the provider network. One use case is integration with 3rd
+party load balancers, where they terminate a load balancer and then send packets directly to OCP nodes with the
+destination IP address being the pod IP itself. Today these load balancers rely on custom operators to detect which node
+a pod is scheduled to and then add routes into its load balancer to send the packet to the right node.
+
+By integrating BGP and advertising the pod subnets/addresses directly on the provider network, load balancers and other
+entities on the network would be able to reach the pod IPs directly.
+
+### Datapath Performance
+
+In cases where throughput is a priority, using the underlay directly can eliminate the need for tunnel encapsulation,
+and thus reducing the overhead and byte size of each packet. This allows for greater throughput.
+
+### Multi-homing, Link Redundancy, Fast Convergence
+
+BGP can use multi-homing with ECMP routing in order to provide layer 3 failover. When a link goes down, BGP can reroute
+via a different path. This functionality can be coupled with BFD in order to provide fast failover.
+
+### User Stories
+
+ * As a user I want to be able to leverage my existing BGP network to dynamically learn routes to pods in my Kubernetes
+   cluster.
+ * As a user, rather than having to maintain routes with NM State manually in each Kubernetes node, as well as being
+   constrained to using local gateway mode for respecting user defined routes; I want to use BGP so that I can dynamically
+   advertise egress routes for the Kubernetes pod traffic in either gateway mode.
+ * As a user where maximum throughput is a priority, I want to reduce packet overhead by not having to encapsulate
+   traffic with Geneve.
+ * As a baremetal or egress IP user, I do not want to have to restrict my nodes to the same layer 2 segment and prefer
+   to use a pure routing implementation to handle advertising virtual IP (VIP) movement across nodes.
+
+### Goals
+
+* To provide a user facing API to allow configuration of iBGP or eBGP peers, along typical BGP configuration including
+  communities, route filtering, etc.
+* Support for advertising Egress IP addresses.
+* To enable BFD to BGP peers.
+* Leveraging BGP to allow for no overlay encapsulation with east/west traffic.
+* ECMP routing support within OVN for BGP learned routes.
+* Support for advertising user-defined networks via BGP as long as there is no subnet overlap.
+
+### Non-Goals
+
+* Support of any other routing protocol.
+* Running separate BGP instances per VRF network.
+* Providing any type of API or operator to automatically connect two Kubernetes clusters via L3VPN.
+* Replacing the support that Metal LB provides today for advertising service IPs.
+* Support for any other type of BGP speaker other than FRR.
+
+### Future Goals
+
+* Support EVPN configuration and integration with a user’s DC fabric, along with MAC-VRFs and IP-VRFs.
```

Comment:
well I view it in the context of with this enhancement we are enabling the multiple networks in OVNK to be advertised outside of the cluster, but only with the default VRF. In the future we are going to add the next stage which will be exposing them per VRF/VPN outside of the cluster.

Reply:
I don't have a strong opinion here.
But we should discuss the "only with the default VRF" part as I think the current design / frr-k8s would allow already to have / advertise isolated networks (see my other comment https://github.com/openshift/enhancements/pull/1636/files#r1634562506)

Reply:
Yep, updated with the vrf-lite option as well that we discussed.